### PR TITLE
Adding deployment delete method

### PR DESCRIPTION
**Pull request by lmanohara**
-Delete deployment related k8 objects.

### CODE REVIEW

File: modules/components/org.wso2.appcloud.provisioning.runtime/src/main/java/org/wso2/appcloud/provisioning/runtime/RuntimeProvisioningService.java [hunk 1]
```diff
@@ -168,4 +168,13 @@ void updateRuntimeProperties(List<RuntimeProperty> runtimeProperty, DeploymentCo
      * @throws RuntimeProvisioningException
      */
     boolean createDeploymentUrl(String environmentUrl) throws RuntimeProvisioningException;
+
+    /**
+     * Delete deployment related K8 objects
+     *
+     * @param deploymentName name of the deployment
+     * @return false if one object deletion is fail
```
Comment (manjulaRathnayaka) [inline modules/components/org.wso2.appcloud.provisioning.runtime/src/main/java/org/wso2/appcloud/provisioning/runtime/RuntimeProvisioningService.java]: Add more descriptive comment about the functionality.
Correct return message with different object it deletes.

File: modules/components/org.wso2.appcloud.provisioning.runtime/src/main/java/org/wso2/appcloud/provisioning/runtime/KubernetesRuntimeProvisioningService.java [hunk 2]
```diff
@@ -828,4 +828,74 @@ public boolean createDeploymentUrl(String environmentUrl) throws RuntimeProvisio
         }
         return created;
     }
+
+    /**
+     * Delete deployment
```
Comment (pirinthapan) [inline modules/components/org.wso2.appcloud.provisioning.runtime/src/main/java/org/wso2/appcloud/provisioning/runtime/KubernetesRuntimeProvisioningService.java]: Put a descriptive method comment. e.g : Delete the deployment in kubernetes.

File: modules/components/org.wso2.appcloud.provisioning.runtime/src/main/java/org/wso2/appcloud/provisioning/runtime/KubernetesRuntimeProvisioningService.java [hunk 3]
```diff
@@ -828,4 +828,74 @@ public boolean createDeploymentUrl(String environmentUrl) throws RuntimeProvisio
         }
```
Comment (pirinthapan) [inline modules/components/org.wso2.appcloud.provisioning.runtime/src/main/java/org/wso2/appcloud/provisioning/runtime/KubernetesRuntimeProvisioningService.java]: There was a seperate code review done for this class and find the below notes and follow them.

1. The method deleteDeployment is lengthy. Break down it into separate methods to delete different kind of objects.
2. The delete method of each object return boolean value. So there may be ‘false’ return even there is no exception. Consider this case as well.
3. Correct the method signature of method deleteDeployment.
4. Some times the pods may not be created properly in kubernetes. Consider this case when deleting pods.
5. Investigate on whether we can use one selector to delete all the objects in kubernetes deployment.
6. There may be null pointer exception for the objects, which are already deleted. Check this and correct the code.
7. The value of applicationContext is retrieved in many places in the same method. Instead initialise it only once.

Comment (lmanohara) [inline modules/components/org.wso2.appcloud.provisioning.runtime/src/main/java/org/wso2/appcloud/provisioning/runtime/KubernetesRuntimeProvisioningService.java]: This is fixed in following commit.
modules/components/org.wso2.appcloud.provisioning.runtime/src/main/java/org/wso2/appcloud/provisioning/runtime/KubernetesRuntimeProvisioningService.java

File: modules/components/org.wso2.appcloud.provisioning.runtime/src/main/java/org/wso2/appcloud/provisioning/runtime/RuntimeProvisioningService.java (hunk 1 above)
Comment (lmanohara) [inline modules/components/org.wso2.appcloud.provisioning.runtime/src/main/java/org/wso2/appcloud/provisioning/runtime/RuntimeProvisioningService.java]: This is fixed in following commit.
modules/components/org.wso2.appcloud.provisioning.runtime/src/main/java/org/wso2/appcloud/provisioning/runtime/KubernetesRuntimeProvisioningService.java

File: modules/components/org.wso2.appcloud.provisioning.runtime/src/main/java/org/wso2/appcloud/provisioning/runtime/KubernetesRuntimeProvisioningService.java (hunk 3 above)
Comment (lmanohara) [inline modules/components/org.wso2.appcloud.provisioning.runtime/src/main/java/org/wso2/appcloud/provisioning/runtime/KubernetesRuntimeProvisioningService.java]: This is fixed in following commit.
https://github.com/wso2/app-cloud/pull/167/commits/5ee0afd6ca0e06f80c511436a7461a9ae5a09b56

**Pull request merged**